packages/controls/src/widget_float.ts

```diff
@@ -85,6 +116,111 @@ class FloatSliderView extends IntSliderView {
     _parse_value = parseFloat;
 }
 
+
+export
+class FloatLogSliderView extends BaseIntSliderView {
+
+    update(options?) {
+        super.update(options);
+        let min = this.model.get('min');
+        let max = this.model.get('max');
+        let value = this.model.get('value');
+        let base = this.model.get('base');
+
+        let log_value = Math.log( value ) / Math.log( base );
+
+        if(log_value > max) {
+            log_value = max;
+        } else if(log_value < min) {
+            log_value = min;
+        }
+        this.$slider.slider('option', 'value', log_value);
+        this.readout.textContent = this.valueToString(value);
+        if(this.model.get('value') !== value) {
+            this.model.set('value', value, {updated_view: this});
+            this.touch();
+        }
+    }
+
+    /**
+     * Write value to a string
+     */
+    valueToString(value: number): string {
+        let format = this.model.readout_formatter;
+        return format(value);
+    }
+
+    /**
+     * Parse value from a string
+     */
+    stringToValue(text: string): number {
+            return this._parse_value(text);
+    }
+
+    /**
+     * this handles the entry of text into the contentEditable label first, the
+     * value is checked if it contains a parseable value then it is clamped
+     * within the min-max range of the slider finally, the model is updated if
+     * the value is to be changed
+     *
+     * if any of these conditions are not met, the text is reset
+     */
+    handleTextChange() {
+        let value = this.stringToValue(this.readout.textContent);
+        let vmin = this.model.get('min');
+        let vmax = this.model.get('max');
+
+        if (isNaN(value as number)) {
```

Comment:
You don't need the `as number` since the `stringToValue` function above is declared as returning a number.